src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2312,11 +2320,19 @@ protected virtual LocalFunctionSymbol LookupLocalFunction(SyntaxToken nameToken)
 
         internal BoundBlock BindEmbeddedBlock(BlockSyntax node, DiagnosticBag diagnostics)
         {
-            return this.GetBinder(node).BindBlock(node, diagnostics);
+            return BindBlock(node, diagnostics);
```

Comment:
I prefer to keep it as we might want to do something special in these scenarios and we already call it where appropriate.